### PR TITLE
[MM-16848] Make Jenkins plugin use a bot account

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes the Jenkins plugin use a bot account to make its posts.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-16848

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -137,15 +161,15 @@ func (p *Plugin) verifyJenkinsCredentials(username, token string) (bool, error)
 
 // createEphemeralPost creates an ephemeral post
 func (p *Plugin) createEphemeralPost(userID, channelID, message string) {
+	botUserID, _ := p.ensureBot()
```
Comment (mickmister) [inline server/plugin.go]: Calling this every time we want to use the bot account, not sure about performance implications. I tried storing it in the `*configuration` struct, but it kept getting overwritten in `onConfigurationChange` / `p.setConfiguration`

Comment (hanzei) [reply server/plugin.go]: Why not store it in the `Plugin` struct?

File: server/plugin.go [hunk 2]
```diff
@@ -158,16 +182,19 @@ func (p *Plugin) createPost(userID, channelID, message string) {
...
 	post := &model.Post{
-		UserId:    userID,
+		UserId:    botUserID,
 		ChannelId: channelID,
 		Type:      model.POST_DEFAULT,
 		Props: map[string]interface{}{
 			"from_webhook":      "true",
-			"override_username": jenkinsUsername,
-			"override_icon_url": p.getConfiguration().ProfileImageURL,
+			"override_username": botUserName,
```
Comment (hanzei) [inline server/plugin.go]: Username and Icon should be handled by the bot.

File: server/plugin.go [hunk 3]
```diff
@@ -158,16 +182,19 @@ func (p *Plugin) createPost(userID, channelID, message string) {
...
+			"override_icon_url": conf.ProfileImageURL,
```
Comment (hanzei) [inline server/plugin.go]: This too.

File: server/plugin.go [hunk 4]
```diff
@@ -3,23 +3,28 @@ package main
...
 	"net/url"
+	"path/filepath"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/model"
+
```
Comment (hanzei) [inline server/plugin.go]: Why is this empty line needed?

File: server/plugin.go [hunk 5]
```diff
@@ -158,17 +187,16 @@ func (p *Plugin) createPost(userID, channelID, message string) {
...
-		UserId:    userID,
+		UserId:    p.botUserID,
 		ChannelId: channelID,
 		Type:      model.POST_DEFAULT,
 		Props: map[string]interface{}{
-			"from_webhook":      "true",
-			"override_username": jenkinsUsername,
-			"override_icon_url": p.getConfiguration().ProfileImageURL,
-			"attachments":       []*model.SlackAttachment{slackAttachment},
+			"from_webhook": "true",
```
Comment (hanzei) [inline server/plugin.go]: Same question

Comment (mickmister) [reply server/plugin.go]: Are you talking about line 190? I don't see a blank line here

File: server/plugin.go [hunk 6]
```diff
@@ -138,14 +169,12 @@ func (p *Plugin) verifyJenkinsCredentials(username, token string) (bool, error)
...
-		UserId:    userID,
+		UserId:    p.botUserID,
 		ChannelId: channelID,
 		Message:   message,
 		Type:      model.POST_DEFAULT,
 		Props: map[string]interface{}{
-			"from_webhook":      "true",
-			"override_username": jenkinsUsername,
-			"override_icon_url": p.getConfiguration().ProfileImageURL,
+			"from_webhook": "true",
```
Comment (hanzei) [inline server/plugin.go]: Why is this Prop needed? It way only needed because of `override_username`, right?

File: server/plugin.go (hunk 5 above)
Comment (hanzei) [reply server/plugin.go]: Missed one comment in my review, sorry. See https://github.com/mattermost/mattermost-plugin-jenkins/pull/15#discussion_r305611578

Comment (mickmister) [reply server/plugin.go]: No worries :slightly_smiling_face:

Thanks for pointing this out @hanzei. I've removed the prop from both places.

File: server/plugin.go (hunk 4 above)
Comment (hanzei) [reply server/plugin.go]: Would you mind fixing this?

**Comment by mickmister** [comment]
Thank you @hanzei

**Pull request merged**